pkg/network/message.go

```diff
@@ -147,11 +159,11 @@ func (m *Message) decodePayload(br *io.BinReader) error {
 
 // Encode encodes a Message to any given BinWriter.
 func (m *Message) Encode(br *io.BinWriter) error {
+	m.tryCompressPayload()
```

Comment:
I think you should either have it here or in `NewMessage`, but not in both.

Reply:
Removed this method from `NewMessage`